lambda-code/nagware/lib/overdueResponseCache.ts

```diff
@@ -0,0 +1,24 @@
+import { RedisConnector } from "./redisConnector.js";
+
+/**
+ * Handles caching of overdue form responses.  The cache expiry should be at least
+ * as long as the interval between Nagware function runs.
+ * 
+ * This cache is used by the app to reduce the number of DynamoDB calls required
+ * when loading the user's `/forms` page. 
+ */
+
+const OVERDUE_CACHE_EXPIRE_SECONDS = process.env.OVERDUE_CACHE_EXPIRE_SECONDS ?? "259200"; // 3 day default
+const OVERDUE_CACHE_KEY = process.env.OVERDUE_CACHE_KEY ?? "overdue:responses:template-ids"
+
+export async function setOverdueResponseCache(
+  formResponses: { formID: string; createdAt: number }[],
+): Promise<void> {
+  const formIDs = formResponses.map((formResponse) => formResponse.formID);
+  const redisConnector = await RedisConnector.getInstance();
+  await redisConnector.client.set(
+    OVERDUE_CACHE_KEY,
+    JSON.stringify(formIDs),
+    { EX: parseInt(OVERDUE_CACHE_EXPIRE_SECONDS, 10) },
```

Comment:
If you agree with my previous comment we could remove the `parseInt` function here.